install/install.class.php

```diff
@@ -515,36 +474,54 @@ public function createNotificationTargetInvitation() {
     *
     * @param string version to upgrade from
     */
-   protected function upgrade($fromSchemaVersion) {
+   public function upgrade(Migration $migration) {
+      spl_autoload_register([__CLASS__, 'autoload']);
+
+      $this->migration = $migration;
+      $fromSchemaVersion = $this->getSchemaVersion();
+
       switch ($fromSchemaVersion) {
          case '0.0':
             // Upgrade to 2.0
-            require_once(__DIR__ . '/upgrade_to_2.0.php');
-            $upgradeStep = new PluginFlyvemdmUpgradeTo2_0();
-            $upgradeStep->upgrade(new Migration('2.0'));
+            $this->upgradeOneStep('2.0');
 
          case '2.0':
             // Example : upgrade to version 2.1
-            // require_once(__DIR__ . '/upgrade_to_2.1.php');
-            // $upgradeStep = new PluginFlyvemdmUpgradeTo2_1();
-            // $upgradeStep->upgrade(new Migration('2.1'));
+            // $this->upgradeOneStep('2.1');
 
          case '3.0':
             // Example : upgrade to version 3.0
-            // require_once(__DIR__ . '/upgrade_to_3.0.php');
-            // $upgradeStep = new PluginFlyvemdmUpgradeTo3_0();
-            // $upgradeStep->upgrade(new Migration('3.0'));
+            // $this->upgradeOneStep('3.0');
 
          default:
             // Must be the last case
             // Unknown version
+            return false;
       }
-      if (is_readable(__DIR__ . '/upgrade_to_develop.php')
-         && PluginFlyvemdmCommon::endsWith(PLUGIN_FLYVEMDM_VERSION, 'develop')
-      ) {
-         $upgradeStep = new PluginFlyvemdmUpgradeToDevelop();
-         $upgradeStep->upgrade(new Migration('develop'));
-      }
+      $this->upgradeOneStep('develop');
```

Comment:
This will make the plugin upgrade always to develop, won't it?

Reply:
Yes, but only if the upgrade_to_develop.php file exists. If it does not exists (this is true in a release) then no problem.